rmw_implementation/src/functions.cpp

```diff
@@ -0,0 +1,513 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstddef>
+#include <cstdio>
+#include <cstring>
+
+#include <list>
+#include <string>
+
+#include <fstream>
+#include <sstream>
+
+#include "Poco/SharedLibrary.h"
+
+#include "rmw/rmw.h"
+
+std::string get_env_var(const char * env_var)
```

Comment:
That's fine, but we should not continue to do this otherwise replacing all instances is just going to get harder and harder.

> Also I don't think rmw is a good location for those since it has nothing to do with the ROS middleware interface.

Well, it does have something to do with it, so far as in the functions are used by many of the rmw implementations (like this meta one). By the same logic, the utilities in rclcpp (like scoped exit) should be in a separate library because it doesn't have anything to do with rclcpp. While that's technically true, it's not very practical.